src/components/input-form/InputForm.component.jsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import InputElem from '../input-elem/InputElem.component';
+import InputSection from '../input-section/InputSection.component';
+import AddSection from '../AddSection/AddSection.component.js';
+
+import './input-form.styles.css';
+
+class InputForm extends React.Component{
+  constructor(props){
+    super(props)
+
+    this.state = {
+      "basics": {
+        "name": "",
+        "label": "",
+        "picture": "",
+        "email": "",
+        "phone": "",
+        "website": "",
+        "summary": "",
+        "location": {
+          "address": "",
+          "postalCode": "",
+          "city": "",
+          "countryCode": "",
+          "region": ""
+        },
+        "profiles": [{
+          "network": "",
+          "username": "",
+          "url": ""
+        }]
+      },
+      "work": [{
+        "company": "",
+        "position": "",
+        "website": "",
+        "startDate": "",
+        "endDate": "",
+        "summary": "",
+        "highlights": []
+      }],
+      "volunteer": [{
+        "organization": "",
+        "position": "",
+        "website": "",
+        "startDate": "",
+        "endDate": "",
+        "summary": "",
+        "highlights": []
+      }],
+      "education": [{
+        "institution": "",
+        "area": "",
+        "studyType": "",
+        "startDate": "",
+        "endDate": "",
+        "gpa": "",
+        "courses": []
+      }],
+      "awards": [{
+        "title": "",
+        "date": "",
+        "awarder": "",
+        "summary": ""
+      }],
+      "publications": [{
+        "name": "",
+        "publisher": "",
+        "releaseDate": "",
+        "website": "",
+        "summary": ""
+      }],
+      "skills": [{
+        "name": "",
+        "level": "",
+        "keywords": []
+      }],
+      "languages": [{
+        "language": "",
+        "fluency": ""
+      }],
+      "interests": [{
+        "name": "",
+        "keywords": []
+      }],
+      "references": [{
+        "name": "",
+        "reference": ""
+      }]
+
+    }
+  }
+
+  // This functino constantly updates the current state of this Class
+  onChangeHandler = (e) => {
+    const { value, name , dirName} = e.target;
+    
+    if (dirName.includes('.')){
```

Comment:
So this is like checking if text is being put in fields? It doest wait for user to stop typing right. Not after every letter. 
Because if it is after every letter it would affect memory.